nvflare/app_common/aggregators/weighted_aggregation_helper.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+from nvflare.apis.dxo import DXO, DataKind, MetaKey, from_shareable
+from nvflare.apis.fl_constant import ReservedKey, ReturnCode
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable
+from nvflare.app_common.abstract.aggregator import Aggregator
+from nvflare.app_common.app_constant import AppConstants
+
+
+class WeightedAggregationHelper(object):
+    def __init__(self, exclude_vars=None, aggregation_weights=None, expected_data_kind=DataKind.WEIGHT_DIFF):
+        """Perform weighted aggregation
+        It computes
+        weighted_sum = sum(dict*weight)
+
+        in accept function
+        The aggregate function returns
+        weighted_sum / sum_of_weights
+
+        Args:
+            exclude_vars ([type], optional): regex to match excluded vars during aggregation. Defaults to None.
+        """
+        super().__init__()
+        self.exclude_vars = re.compile(exclude_vars) if exclude_vars else None
+        self.reset_stats()
+        self.total = dict()
+        self.counts = dict()
+        self.history = list()
+
+    def reset_stats(self):
+        self.total = {}
+        self.counts = {}
+        self.history = []
+
+    def add(self, data, weight, client_name, contribution_round) -> bool:
```

Comment:
Suggest change client_name to contributor_name to be more generic.

Reply:
done